### PR TITLE
Learning rate schedules, ignore padding in loss and add smoke tests

### PR DESCRIPTION
**Pull request by LoicGrobol**
The main motivation for this was to be able to use the linear decay with warmup LR schedule used in most of the BERT fine-tuning litterature. It gives the ability to chose a LR decay shape: exponential with a 0.95 decay rate (default and the previous behaviour), linear or constant, and a number of linear warmup steps (defaulting to 0). Along the way I noticed that the CrossEntropyLoss was taking into account the performances on the padding tokens (which I removed) and I finally added some smoke test that can be run either in local or in github CI and which ensures that we are at least able to learn and parse a toy treebank and eval the results without crashing.

Overall, this has no groundbreaking impact on scores. It does make the parser learn really faster (in terms of epochs) and reduce the computation and memory load.

## Added

- A new config option `lr_schedule` of the form
  ```yaml
  lr_schedule:
    shape: linear
    warmup_steps: 100
  ```
  which allows to chose a decay shape and a number of linear warmup steps
- Examples of training configurations (in `examples/`)
- Smoke tests using tox and used as checks in Github CI to ensure that `graph_parser` at least runs without crashing and outputs a valid model and valid parses. They will also auto-release to Pypi when we create new releases on Github if we set up a Pypi project and add the credentials as repository secrets.

## Changed

- Padding tokens are now ignored in the training and development losses

### CODE REVIEW

**Comment by bencrabbe** [comment]
just a question: the embedding layers and the losses where set with a padding_idx and ignore_idx arguments.
that were supposed to ignore the padding stuff.  Do you mean that it was not used at all ?

**Comment by LoicGrobol** [comment]
The embeddings did use padding_idx but the loss function [currently](hhttps://github.com/bencrabbe/npdependency/blob/b646c45e7e9498ba259ba53236085912744a2775/npdependency/graph_parser.py#L296) uses default `ignore_index` (-100) which was not the padding used for the labels so the padding labels did contribute to the loss. Also I think setting the `ignore_index` to the 0 used for label padding would be an issue since it is also the index of the root (which should obviously not be ignored).

**Comment by LoicGrobol** [comment]
I did some archaeology and apparently the [first version of the graph parser](https://github.com/bencrabbe/npdependency/blob/50d3538e282117a9f9af3b4af83e723d98468c46/src/graph_parser.py#L244) used `ignore_index` but it was lost during the switch to `graph_parser2` :-)
